sarif/src/main/java/com/jetbrains/qodana/sarif/model/MapIgnoreKeysAdapter.java

```diff
@@ -0,0 +1,73 @@
+package com.jetbrains.qodana.sarif.model;
+
+import com.google.gson.*;
+import com.google.gson.internal.bind.ObjectTypeAdapter;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+import java.util.*;
+
+public class MapIgnoreKeysAdapter extends TypeAdapter<Map<String, Object>> {
+
+    private static final Gson embedded = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
```

Comment:
Hm, I think it makes more sense to pass configured gson instances via constructor (here, and everywhere else with an embedded gson), for two reasons:
1. Serialization Correctness: Let's say we have a property bag that  contains `java.lang.Instant`: If encountered in other fields, they will be serialized according to spec (see `IsoInstantSerializer`), but if they are encountered in property bag, it would use gson's default.
While the spec doesn't necessarily say how custom values in properties are to be serialized, it's at least very confusing to have potentially different formats in the same file
2. Performance: Gson caches how to create types, and these caches could be shared between our type adapters (`Gson#constructorConstructor`)

IIRC the `TypeAdapterFactory` approach would help us do that

Reply:
Yes, we would have to implement a factory, since otherwise we don't have access to gson instance, since type adapte is constructed before, I can do that. 

Bit I'm not convinced that we want this adapter for all classes though. We only want to ignore properties inside PropertyBag, if I create a factory, and refactor to use a single gson instance, gson will use it for all types assignable to Map<String,Object>. If I instantiate a new gson instance just in PropertyBag with a factory, we don't gain much and have added complexity in factory code

Reply:
Actually, I can just create a type that wraps map and use a factory with that. Will do

Reply:
Actually, why even split the property adapter from the ignore keys adapter? If they are one, we don't have that problem, as that single adapter would only be registered for `PropertyBag`

Reply:
Hm, actually I like the idea of keeping it all in the property bag adapter. I thought it's cleaner to keep it separate, but the logic inside property bag adapter is very minimal so we could merge it with reading typeless map